src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,177 @@
+package edu.wpi.grip.core;
+
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Point;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+
+/**
+ * Commonly used {@link SocketHint SocketHints}
+ */
+public final class SocketHints {
+
+    /**
+     * Provides some default domains
+     */
+    public enum Domain {
+        INTEGERS(Integer.MIN_VALUE, Integer.MAX_VALUE),
+        POSITIVE_INTEGERS(0d, Integer.MAX_VALUE),
```

Comment:
0 isn't positive.

This is another disadvantage of this - someone might use `POSITIVE_INTEGERS` and expect it to start at 1.  Even if it's fixed, I'm sure plenty of people might be confused at why it doesn't start at 0.